### ISSUE TITLE
CPP version checks fail for unpacked thunks loaded into 'ob watch'

### ISSUE THREAD

**Issue by alexfmpe** [obsidiansystems/obelisk#756]
To reproduce: https://github.com/obsidiansystems/obelisk/compare/develop...alexfmpe:ae@reproduce-unpacked-cpp

Works in `cabal new-repl` but not `ob watch`. 
Seems like the macros for unpacked thunks aren't being loaded.

**Comment by 3noch** [comment]
Obelisk doesn't define these macros so until it does you need to wrap CPP like this in `#if defined(MIN_VERSION_obelisk_route)`

**Comment by 3noch** [comment]
Of course, if you want the stuff inside the CPP to work, then you might have to restructure it. It's really not ideal.

**Comment by madeline-os** [comment]
Is this still an issue?

**Comment by alexfmpe** [comment]
> Obelisk doesn't define these macros 

Who does then? `nix-build -A ghc.frontend` works just fine.

> Is this still an issue?

I merged the branch in the original post with develop and still reproduce the error message.
By now I forgot the situation where this arose, so I don't have a motivating example, but given it's an inconsistency with the "real build", I'd say so.

**Comment by 3noch** [comment]
Cabal generates and includes these during build

**Comment by madeline-os** [comment]
:wave:
